### PR TITLE
use custom modulemap

### PR DESCRIPTION
**Pull request by sushichop**
### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/CocoaLumberjack/)
* [x] I have searched for a similar pull request in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/pulls) and found none

* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [ ] I have added the required tests to prove the fix/feature I am adding
* [ ] I have updated the documentation (if necesarry)
* [x] I have run the tests and they pass
* [x] I have run the lint and it passes (`pod lib lint`)

This merge request reffers to the following issue: #815 
This merge request reffers to the following commit: https://github.com/CocoaLumberjack/CocoaLumberjack/commit/0f5a793843c748c06511e5c3d1e20967798b50d1

### Pull Request Description

Custom module map was removed by this commit(https://github.com/CocoaLumberjack/CocoaLumberjack/commit/0f5a793843c748c06511e5c3d1e20967798b50d1) and it should be a temporizing workaround. This PR enables custom modulemap again as CocoaLumberjack used it before(e.g. CocoaLumberjack 2.3.0 used it.).

However, keep in mind. 
In order to do this, podspec will be summarized to 3 subspecs for the reason why latest CocoaPods (probably from version 1.0.0) requires the strict podspec. 

For example,` DDDispatchQueueLogFormatter` is declared explicitly in the CocoaLumberjack.modulemap file, on the other hand, the original several subspecs(`Core` or `Extensions`) does not include this header. In this case `pod lib lint` fails. In other words, it is a root cause.
Moreover, as a podspec syntax, you can not specify `module_map` attribute in subspec level.
So I needed to change podspec.

Thanks.

### CODE REVIEW

File: CocoaLumberjack.podspec [hunk 1]
```diff
@@ -24,22 +24,16 @@ Pod::Spec.new do |s|
   s.watchos.deployment_target = '2.0'
   s.tvos.deployment_target    = '9.0'
 
-  s.default_subspecs = 'Default', 'Extensions'
+  s.module_map = 'Framework/Lumberjack/CocoaLumberjack.modulemap'
+  s.default_subspecs = 'Default'
```
Comment (rivera-ernesto) [inline CocoaLumberjack.podspec]: This may not be needed anymore but I'm not sure about losing the other subspecs.

Comment (sushichop) [inline CocoaLumberjack.podspec]: Thank you for your comment.

**s.module_map**

If you do not have a plan to support backward compatibility, 
`s.module_map` is not needed anymore.
In this case, backward compatibility means the followings.

When user uses `DDDispatchQueueLogFormatter` class
 [CocoaLumberjack 2.3]
import CocoaLumberjack.DDDispatchQueueLogFormatter
[CocoaLumberjack latest]
import CocoaLumberjack

To be honest, I also think cusutom modulemap is not needed anymore as you say😁
Without custom modulemap, CocoaPods automatically makes a normal modulemap file.
I made a PR to discuss about it.

**s.default_subspecs**
On the other hand, `s.default_subspecs` should be needed.
Because if this setting is removed from podspec, Swift Standard Libraries are always included even when user wants to use only Objective-C code.

To sum it up, I'll be gladly to decline(close) this PR if you do not have a plan to support backward compatibility about custom modulemap.

BTW, if you do not need custom modulemap, I think probably you had better change the modulemap of Lumberjack.xcodeproj a little. Each target still uses custom modulemap file that is not different from a modulemap file CocoaPods automatically makes.
Can I make an another PR to do this?

Comment (rivera-ernesto) [inline CocoaLumberjack.podspec]: Yes please. I think that at this point backwards compatibility with 2.x.x is less important than having cleaner project.

Comment (sushichop) [inline CocoaLumberjack.podspec]: Good! Then I'm closing this PR in first. About another PR..., please wait for a little while? (probably at the longest less than 1 week though)
Thanks!

Comment (sushichop) [inline CocoaLumberjack.podspec]: @rivera-ernesto 

> I think probably you had better change the modulemap of Lumberjack.xcodeproj a little.

I checked deeply and **actually** each target's headers and modulemap. As a result  I found no need change... Sorry for confusing you...
But if I find another issue or come to better solution, I'd like to discuss or make a PR.
Thanks again!!
